src/main/java/org/primeframework/mvc/security/CipherProvider.java

```diff
@@ -22,13 +22,13 @@
 import java.security.NoSuchAlgorithmException;
 
 /**
- * Provider for getting a Cipher instance that can be used for encryption and any other security as necessary.
+ * Provider for getting a Cipher instance that can be used for AES encryption and any other security as necessary.
  *
  * @author Brian Pontarelli
  */
 public interface CipherProvider {
   /**
-   * Return a cipher used for decrypting.
+   * Return a cipher used for decrypting AES.
```

Comment:
We shouldn't describe the implementation in the interface. The interface only provides a cipher of unknown implementation.

Reply:
Point taken, but the `iv` parameter implies certain things about the cipher returned. Not all ciphers require an initialization vector.